erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -253,17 +253,24 @@ def get_directions(self, route, optimize):
 
 
 @frappe.whitelist()
-def get_delivery_window(customer=None):
-	delivery_window = frappe.db.get_value("Customer", customer, ["delivery_start_time", "delivery_end_time"], as_dict=1)
-	delivery_window.default_window = False
-
-	if delivery_window and (delivery_window.delivery_start_time or delivery_window.delivery_end_time):
+def get_delivery_window(doctype=None, docname=None, customer=None):
+	if customer:
+		delivery_window = frappe.db.get_value("Customer", customer, ["delivery_start_time", "delivery_end_time"], as_dict=1)
+		delivery_window.default_window = False
+
+		if delivery_window and (delivery_window.delivery_start_time or delivery_window.delivery_end_time):
+			return delivery_window
+		
+		default_window = frappe.db.get_value("Delivery Settings", "Delivery Settings", ["delivery_start_time", "delivery_end_time"], as_dict=1)
```

Comment:
Move this into the `else` for this condition. The system should always fallback to the times defined in Delivery Settings if no value is given to this function.